### PR TITLE
Upgrade canal

### LINKED ISSUE #6016: Upgrade Canal as a CNI for user clusters.
**User Story**
We are currently using Canal as a CNI for user clusters. Till we implement #5589 and for clusters running older versions of Kubermatic, we should upgrade to the latest version: 3.16.3 from 3.8.0.

**Acceptance criteria**
- Canal in VXLAN mode is installed in user clusters.
- Having upgrade documentation for existing user clusters.

### PR DESCRIPTION
**Pull request by irozzo-1A**
**What this PR does / why we need it**:
This PR upgrades canal addon.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #6016 

**Special notes for your reviewer**:
The old manifest was not removed from the addon to avoid forcing migration of already existing clusters, as the upgrade path has to be tested carefully. The upgrade is out of scope from this PR.

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by mlavacca** [comment]
Fixes #7180

File: pkg/crd/kubermatic/v1/cluster.go [hunk 1]
```diff
@@ -61,6 +61,11 @@ const (
 	MaxEtcdClusterSize     = 9
 )
 
+const (
+	AnnotationCNIPluginKind    = "cni-plugin.k8c.io/kind"
```
Comment (rastislavs) [inline pkg/crd/kubermatic/v1/cluster.go]: I may be missing something, but I do not see where these annotations are being used

File: pkg/webhook/cluster/mutation/mutation.go [hunk 2]
```diff
@@ -96,6 +96,15 @@ func (h *AdmissionHandler) Handle(ctx context.Context, req webhook.AdmissionRequ
 }
 
 func (h *AdmissionHandler) applyDefaults(c *kubermaticv1.Cluster) {
+	// Add cni-plugin annotation if not present pointing to latest canal
```
Comment (rastislavs) [inline pkg/webhook/cluster/mutation/mutation.go]: actually this does not seem to be adding annotation as the comment suggests, but modifying the cluster spec

File: pkg/crd/kubermatic/v1/cluster.go (hunk 1 above)
Comment (irozzo-1A) [reply pkg/crd/kubermatic/v1/cluster.go]: Good catch :-) I started using the annotations then I thought that extending the cluster CRD was cleaner.

Comment (rastislavs) [reply pkg/crd/kubermatic/v1/cluster.go]: yeah I thought so :)

File: pkg/crd/kubermatic/v1/datacenter.go [hunk 3]
```diff
@@ -152,7 +152,7 @@ type DatacenterSpec struct {
 
 	//nolint:staticcheck
 	//lint:ignore SA5008 omitgenyaml is used by the example-yaml-generator
-	Fake *DatacenterSpecFake `json:"fake,omitempty,omitgenyaml"`
+	Fake *DatacenterSpecFake `json:"fake,omitempty,omitempty"`
```
Comment (wozniakjan) [inline pkg/crd/kubermatic/v1/datacenter.go]: nit: looks like duplicated `omitempty` parameter, is that intentional?

Comment (irozzo-1A) [reply pkg/crd/kubermatic/v1/datacenter.go]: nope, this is totally unintentional. Thx for pointing out ;-)

**Pull request merged**